### ISSUE TITLE
 Feature Request: Export CSS for Sennheiser WSM Software Compatibility

### ISSUE THREAD

**Issue by leo-benjamin** [berkon/wireless-microphone-analyzer#43]
First of all, thank you so much for this software. With the added RF Explorer Plus compatibility, it perfectly fits my needs. Since I notice you are actively developing this software, I wanted to request a feature.

When exporting the CSV, could you please add an extra option for compatibility with Sennheiser's Wireless Systems Manager (WSM) software? Specifically, WSM software requires:

Whole numbers to be divided by a comma (",") instead of a period (".").
Values to be separated by a semicolon (";") instead of a comma (",").
This feature would greatly enhance the usability of the software for those of us using Sennheiser's WSM.

Thank you for considering this request!

**Comment by berkon** [comment]
Hi @leo-benjamin, thanks for your request. I'm very happy to hear that my tool is useful for you! :-) I don't have Sennheiser's Wireless Systems Manager installed and I don't know if it can be used for free ... for testing purpose. But if the change is really that simple, it should not be a problem. I'll have a look and will let you know. ;-)

**Comment by berkon** [comment]
@leo-benjamin, I've just installed version 4.8.0 (Build 16) of the Sennheiser WSM tool and when I go to Frequency Manager > Professional Setup and click the Import button, it looks like its importing the current Shure CSV without any issue:

![image](https://github.com/berkon/wireless-microphone-analyzer/assets/28842661/ee0df4ac-befd-47b7-9f60-9f9b793399bd)

**Comment by leo-benjamin** [comment]
Hi,

Thank you for responding and testing. I have updated my WSM software, and it loads the CSV file, but it's not accurate. Here are the steps I follow to reproduce the issue:

1. Open the wireless-microphone-analyzer.
2. Use band preset 1 (510KHz-560KHz).
3. Set Sweeppoints to 5000.
4. Export the trace.
5. Make a copy of this file and change all commas (",") to periods (".") and all commas (",") to semicolons (";").

If I now import the original CSV file:
[Original scan file.csv](https://github.com/user-attachments/files/15897463/Original.scan.file.csv)
I get the following result:
<img width="787" alt="Original file result" src="https://github.com/berkon/wireless-microphone-analyzer/assets/120786735/8f2293f1-933a-457e-86a5-f0e2a0df3b7f">
Notice that the accuracy of this scan is not that high, and around the 560KHz point, suddenly things change.

If I import the changed CSV file:
[Changed scan file.csv](https://github.com/user-attachments/files/15897486/Changed.scan.file.csv)
I get the following result:
<img width="788" alt="Changed file result" src="https://github.com/berkon/wireless-microphone-analyzer/assets/120786735/2a75f131-83ca-4935-8a40-c33be59d6249">
This looks to me like the correct scan data with the 5000 sweep points.

Am I missing something here? Am I doing something wrong?
For reference, I am using a MacBook Pro M3.

Thank you for your assistance.

**Comment by berkon** [comment]
Thanks a lot for sending all the details. There are a few things which look strange to me regarding the WSM Tool.

I tried to export the imported data in order to see how the exact format must look like. But for some reason it looks like its converting dbm to db (which in fact is not possible to my knowledge) in the exported file. So a value like -77 in the imported file becomes 30 in the exported file. This correlates to switching between dbm and db in the WSM UI. But as said, at least to my knowledge this conversion is not possible. Interestingly it is obviously able to detect that the values are in dbm and import them correctly, even though (according to the export) it seem to prefer db values. Another thing is that the export contains two semicolons (';;') between the frequency and the value.

Long story short: I started a request to Sennheiser, asking them to provide details regarding the CSV format they expect for import. I think that's better than doing reverse engineering. If they don't respond, I will implement it according to the information we have gathered so far.

**Comment by leo-benjamin** [comment]
Once again, thank you for being so quick and thorough on your end.

In my opinion, the Sennheiser software still has many flaws, and I would much rather use Shure's WWB7. However, I mostly work with Sennheiser IEMs, so I rely on WSM for its quicker functionality (if only it worked properly).

Thanks again for your help and for reaching out to Sennheiser for further clarification.

**Comment by berkon** [comment]
@leo-benjamin, I just release V2.2.0. There is now an option to export for Sennheiser. Take a look ...

**Comment by leo-benjamin** [comment]
Hi Bernd

This absolutely amazing thank you so much.
You are the best!

Kind regards
Leo-Benjamin

Op 1 jul 2024, om 13:01 heeft Bernd Konnerth ***@***.***> het volgende geschreven:



@leo-benjamin<https://github.com/leo-benjamin>, I just release V2.2.0. There is now an option to export for Sennheiser. Take a look ...

**Comment by berkon** [comment]
Thank you very much! You're welcome! ;-)
